addons/shortcodes.php

```diff
@@ -59,179 +77,280 @@ function shutdown() {
+		);
+		$posts  = get_posts( $params );
 
-		// Enqueue Jetpack's spinner script if available
-		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
-			wp_enqueue_script( 'jquery.spin' );
+		if ( ! $posts ) {
+			return;
 		}
 
-		// For wp.template()
-		wp_enqueue_script( 'wp-util' );
+		$regex = get_shortcode_regex( array( 'camptix_attendees' ) );
 
-		// Lazy load the camptix js.
+		foreach ( $posts as $post ) {
+			$matches = array();
+
+			if ( ! preg_match_all( $regex, $post->post_content, $matches, PREG_SET_ORDER ) ) {
+				continue;
+			}
+
+			foreach ( $matches as $match ) {
+				$attr = shortcode_parse_atts( $match[3] );
+				$attr = $this->sanitize_attendees_atts( $attr );
+
+                $this->get_attendees_shortcode_content( $attr );
+			}
+		}
+	}
+
+	/**
+	 * Callback for the [camptix_attendees] shortcode.
+	 */
+	public function shortcode_attendees( $attr ) {
+		// Required scripts
+		wp_enqueue_script( 'wp-util' ); // For wp.template()
+		if ( wp_script_is( 'jquery.spin', 'registered' ) ) {
+			wp_enqueue_script( 'jquery.spin' ); // Enqueue Jetpack's spinner script if available
+		}
 		wp_enqueue_script( 'camptix' );
 
 		// Only print the JS template once.
 		if ( ! has_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) ) ) {
 			add_action( 'wp_print_footer_scripts', array( $this, 'avatar_js_template' ) );
 		}
 
-		$start = microtime(true);
+		$attr = $this->sanitize_attendees_atts( $attr );
+
+		/**
+		 *
+		 */
+		do_action( 'camptix_attendees_shortcode_init', $attr );
+
+		return $this->get_attendees_shortcode_content( $attr );
+	}
+
+	/**
+	 * Generate the key for a particular configuration to use when
+	 * setting or retrieving a cached value.
+	 *
+	 * @param array $attr    Sanitized shortcode attributes
+	 *
+	 * @return string        The cache key
+	 */
+	protected function generate_attendees_cache_key( $attr ) {
+		return 'camptix-attendees-' . md5( maybe_serialize( $attr ) );
+	}
+
+	/**
+	 * Get the content for an instance of the [camptix_attendees] shortcode.
+	 *
+	 * This checks for a cached version first. If none is found, it generates
+	 * the content and caches it before returning.
+	 *
+	 * @param array $attr             Sanitized shortcode attributes
+	 * @param bool  $force_refresh    True to generate the content even if cached value is found.
+	 *
+	 * @return string                 Rendered shortcode content
+	 */
+	public function get_attendees_shortcode_content( $attr, $force_refresh = false ) {
+		global $camptix;
+
+		// Cache duration. Day for active sites, month for archived sites.
+		$camptix_options = $camptix->get_options();
+		$cache_time = ( $camptix_options['archived'] ) ? DAY_IN_SECONDS * 30 : DAY_IN_SECONDS;
 
-		// Serve from cache if cached copy is fresh.
-		$transient_key = md5( 'tix-attendees' . serialize( $attr ) );
-		if ( false !== ( $cached = get_transient( $transient_key ) ) ) {
-			if ( ! is_array( $cached ) )
-				return $cached; // back-compat
```

Comment:
I figured since the cache key is changing, there's no way it would retrieve an old transient that wasn't an array, so the backcompat would no longer be necessary. I'll add a note.